app/components/Map.js

```diff
@@ -13,8 +15,48 @@ export default class Map extends Component {
       position: null,
       selectedPin: undefined,
       dropPinLocation: undefined,
+      loaded: false,
+      friendLocs: {},
     };
   }
+// Friend
+  componentWillMount() {
+    const { friends } = this.props;
+    let self = this;
+    let counter = 0;  
+     for(var friendId in friends) {
+      self.setListener(friends[friendId]);
+      counter++;
+      if(counter === Object.keys(friends).length) {
+        this.setState({loaded: true});
+      }
+    }
```

Comment:
`friends` resolves to an empty object on first load, preventing 'loaded' from ever becoming true